IPython/core/magics/basic.py

```diff
@@ -377,7 +379,24 @@ def xmode_switch_err(name):
             xmode_switch_err('user')
 
     @line_magic
-    def quickref(self,arg):
+    def pip(self, args=''):
+        """
+        Intercept usage of `pip` in the IPython shell and print an explanation.
+        """
+        print(textwrap.dedent('''
+        The following command is not meant to be run from a Python shell:
```

Comment:
Also did you mean changing the first line all the all text for your alternative ?